Update RateLimiter to use dictionary of rate limits (fixes #437)

MingweiSamuel opened the pull request:
![image](https://user-images.githubusercontent.com/6778341/28199717-adc715ee-681c-11e7-8379-f2989c8a3611.png)

Tests pass locally (don't have tournament key). Not sure if CI is updated for the new system so...

BenFradet commented on RiotSharp.Test/appsettings.json:
```diff
@@ -1,4 +1,4 @@
 ﻿{
-  "ApiKey": "API_KEY",
+  "ApiKey": "RGAPI-369f27ac-9250-4e30-95e1-6fa1477f0048",
```
!!

MingweiSamuel replied:
Lul

JanOuborny commented on RiotSharp.AspNetCore/RiotSharpOptions.cs:
```diff
@@ -9,16 +9,12 @@ public class RiotSharpOptions
             RiotApi = new ApiKeyOptions();
-            RiotApi.RateLimitPer10S = 10;
-            RiotApi.RateLimitPer10M = 500;
             TournamentApi = new TournamentApiKeyOptions();
```
Can you add default values here?

MingweiSamuel commented:
Removed the api key, added default RiotApi rate limits for the options. I don't have any experience with the Tournament API so I don't how those rate limits are. I figure that API isn't meant to be spammed anyway.

JanOuborny commented:
@MingweiSamuel  The default rate limits for my tournament key are 10 per 10 seconds and 500 per 10 minutes. Can you also add these?

MingweiSamuel commented:
Updated

BenFradet commented:
Merging, thanks!
